### PR TITLE
fix: wire up nameTransformer in secret processors

### PR DESCRIPTION
**Pull request by nimish-ks**
## Summary

- **Fix `nameTransformer` not being applied**: The `nameTransformer` parameter was accepted by `process_secrets()` but never used — only `asName` worked. This wires up `transform_name()` with proper priority: `asName` > per-key `nameTransformer` > secret-level `nameTransformer`.
- **Remove `dotnet-env` transformer**: Was buggy with double underscores (`DB__USER_NAME` → `Db____User__Name`). Users can use `asName` for explicit renaming instead.
- **Rename `src/cmd/` to `src/internal/`**: Avoids shadowing Python's stdlib `cmd` module.
- **Add `nameTransformer` to CRD**: Documented at the `managedSecretReferences` level so users can apply a default transform to all keys.
- **Add 81 unit tests**: Comprehensive coverage for `process_secrets()` and `transform_name()` including edge cases.
- **Add CI workflow**: Python unit tests now run on PRs and pushes that touch `src/`.
- **Bump version to 1.4.1**

## Related

- Docs PR: https://github.com/phasehq/docs/pull/205
- Linear: https://linear.app/phasehq/issue/PH-596/qa-nametransformer-fix-in-secret-processors

## Context

Reported by a user — after migrating to Phase, `nameTransformer` declarations (e.g. `upper-camel`) had no effect while `asName` worked fine. Root cause: `transform_name()` existed in `utils/misc.py` but was never called from `process_secrets()`. This was a regression from when secret processing moved client-side.

## Test plan

- [x] 81 unit tests passing locally (`pytest -v`)
- [x] CI workflow runs on this PR
- [x] Manual test with kopf dev server — verified `nameTransformer: camel`, per-key `upper-camel`, and `asName` all work correctly

### CODE REVIEW

**Review by rohan-chaturvedi — APPROVED** [review]

**Review by rohan-chaturvedi — APPROVED** [review]

File: phase-kubernetes-operator/crds/crd-template.yaml [hunk 1]
```diff
@@ -96,6 +96,15 @@ spec:
...
                           - "bootstrap.kubernetes.io/token"
+                      nameTransformer:
+                        description: Default name transformer applied to all secret keys. Per-key nameTransformer in processors takes precedence.
+                        type: string
+                        enum:
+                          - "camel"
+                          - "upper-camel"
+                          - "lower-snake"
+                          - "tf-var"
+                          - "lower-kebab"
```
[the remark this answers is not in the document]
Comment (nimish-ks) [reply phase-kubernetes-operator/crds/crd-template.yaml:107]: removed dotnet-env.

File: src/main.py [hunk 2]
```diff
@@ -205,8 +215,7 @@ def process_secrets(fetched_secrets, processors, secret_type, name_transformer):
             # Default to plain processing if processor type is not recognized
             processed_value = base64.b64encode(value.encode()).decode()
 
-        # Map the processed value to the asName
-        processed_secrets[as_name] = processed_value
+        processed_secrets[output_key] = processed_value
```
[the remark this answers is not in the document]
Comment (nimish-ks) [reply src/main.py:230]: Added logging.

**Review by rohan-chaturvedi — APPROVED** [review]
